### PR TITLE
Mech Repair Now Fumbles

### PR DESCRIPTION
**Pull request by Ben10083**
## About The Pull Request
Mech Repairs now have a skill check, failing it leads to fumble.

## Why It's Good For The Game
I am an undercover agent dispatched by the xeno mains to nerf all fun marines have, I will now be making it so the private can't repair the mech as easily as the synth. 

## Changelog
:cl:
balance: Fumble added to mech repairs.
/:cl:

### CODE REVIEW

File: code/modules/vehicles/mecha/mecha_defense.dm [hunk 1]
```diff
@@ -252,6 +252,13 @@
 		return
 	if(!W.tool_start_check(user, amount=1))
 		return
+	//check if a dumbass
+	var/skill = user.skills.getRating("engineer")
+	if(skill < SKILL_ENGINEER_ENGI) //Better be at least a engineer or you going to have a bad time
+		user.visible_message(span_notice("[user] fumbles around figuring out how to use [W]."),
+		span_notice("You fumble around figuring out how to use [W]."))
+		if(!do_after(user, 150 * (SKILL_ENGINEER_ENGI - skill), TRUE, user, BUSY_ICON_UNSKILLED))
+			return
```
Comment (TiviPlus) [inline code/modules/vehicles/mecha/mecha_defense.dm:261]:
```suggestion
		if(!do_after(user, 30 * (SKILL_ENGINEER_ENGI - skill), TRUE, user, BUSY_ICON_UNSKILLED))
			return
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by Ben10083** [comment]
Talked with Tivi over discord, fumble time is now 30 ticks and fumble occur for each repair.

File: code/modules/vehicles/mecha/mecha_defense.dm [hunk 2]
```diff
@@ -255,7 +255,13 @@
 	user.balloon_alert_to_viewers("started welding [src]", "started repairing [src]")
 	audible_message(span_hear("You hear welding."))
 	var/did_the_thing
+	var/skill = user.skills.getRating("engineer") //Dumbass check
 	while(obj_integrity < max_integrity)
+		if(skill < SKILL_ENGINEER_ENGI) //fumble for every repair
+			user.visible_message(span_notice("[user] fumbles around figuring out how to repair [src]."),
+			span_notice("You fumble around figuring out how to repair [src]."))
+			if(!do_after(user, 30 * (SKILL_ENGINEER_ENGI - skill), TRUE, user, BUSY_ICON_UNSKILLED)) //repeat for different text for ongoing repairs
```
Comment (TiviPlus) [inline code/modules/vehicles/mecha/mecha_defense.dm:263]:
```suggestion
	var/skill = user.skills.getRating("engineer")
	while(obj_integrity < max_integrity)
		if(skill < SKILL_ENGINEER_ENGI)
			user.visible_message(span_notice("[user] fumbles around figuring out how to repair [src]."),
			span_notice("You fumble around figuring out how to repair [src]."))
			if(!do_after(user, 30 * (SKILL_ENGINEER_ENGI - skill), TRUE, user, BUSY_ICON_UNSKILLED))
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/vehicles/mecha/mecha_defense.dm [hunk 3]
```diff
@@ -255,7 +255,13 @@
 	user.balloon_alert_to_viewers("started welding [src]", "started repairing [src]")
 	audible_message(span_hear("You hear welding."))
 	var/did_the_thing
+	var/skill = user.skills.getRating("engineer")
 	while(obj_integrity < max_integrity)
+		if(skill < SKILL_ENGINEER_ENGI)
+			user.visible_message(span_notice("[user] fumbles around figuring out how to repair [src]."),
+			span_notice("You fumble around figuring out how to repair [src]."))
```
Comment (TiviPlus) [inline code/modules/vehicles/mecha/mecha_defense.dm:262]: use a balloon alert instead

**Review by TiviPlus — APPROVED** [review]
Rest fine

File: code/modules/vehicles/mecha/mecha_defense.dm [hunk 4]
```diff
@@ -252,10 +252,18 @@
...
 	if(!W.tool_start_check(user, amount=1))
 		return
-	user.balloon_alert_to_viewers("started welding [src]", "started repairing [src]")
-	audible_message(span_hear("You hear welding."))
 	var/did_the_thing
+	var/started = FALSE
+	var/skill = user.skills.getRating("engineer")
 	while(obj_integrity < max_integrity)
+		if(skill < SKILL_ENGINEER_ENGI)
+			user.balloon_alert_to_viewers("[user] fumbles around figuring out how to repair [src].")
```
Comment (TiviPlus) [inline code/modules/vehicles/mecha/mecha_defense.dm:260]:
```suggestion
			user.balloon_alert_to_viewers("fumbles around trying to repair")
```

File: code/modules/vehicles/mecha/mecha_defense.dm [hunk 5]
```diff
@@ -252,10 +252,18 @@
...
+			if(!do_after(user, 30 * (SKILL_ENGINEER_ENGI - skill), TRUE, user, BUSY_ICON_UNSKILLED))
+				return
+		if(!started)
+			user.balloon_alert_to_viewers("started welding [src]", "started repairing [src]")
```
Comment (TiviPlus) [inline code/modules/vehicles/mecha/mecha_defense.dm:264]:
```suggestion
			user.balloon_alert_to_viewers("started welding", "started repairing")
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Since balloon alerts are onmap they should be short which means no name inserts usually

**Pull request merged**